doc/source/querying.rst

```diff
@@ -0,0 +1,96 @@
++----------------+-----------------------------------------------------------------------+-----------+---------------------------+---------------------------------------------------------------------+
+| r_min          | Minimum distance to find neighbors                                    | float     | 0 <= r_min < r_max        | :class:`freud.locality.LinkCell`, :class:`freud.locality.AABBQuery` |
++----------------+-----------------------------------------------------------------------+-----------+---------------------------+---------------------------------------------------------------------+
+| num_neighbors  | Number of Neighbors                                                   | int       | num_neighbors >= 0        | :class:`freud.locality.LinkCell`, :class:`freud.locality.AABBQuery` |
++----------------+-----------------------------------------------------------------------+-----------+---------------------------+---------------------------------------------------------------------+
+| exclude_ii     | Whether or not to include self-neighbors                              | bool      | True/False                | :class:`freud.locality.LinkCell`, :class:`freud.locality.AABBQuery` |
++----------------+-----------------------------------------------------------------------+-----------+---------------------------+---------------------------------------------------------------------+
+| r_guess        | Initial search distance for sequence of ball queries                  | float     | r_guess > 0               |  :class:`freud.locality.AABBQuery`                                  |
++----------------+-----------------------------------------------------------------------+-----------+---------------------------+---------------------------------------------------------------------+
+| scale          | Amount to increase r_guess when not enough neighbors are found        | float     | scale > 1                 |  :class:`freud.locality.AABBQuery`                                  |
++----------------+-----------------------------------------------------------------------+-----------+---------------------------+---------------------------------------------------------------------+
+
+
+Mode Deduction
+==============
+
+The ``mode`` query argument specifies the type of query that is being performed, and it therefore governs how other arguments are interpreted.
+In most cases, however, the query mode can be deduced from the set of query arguments.
+Specifically, any query with the ``num_neighbors`` key set is assumed to be a query with ``mode=nearest``.
+For completeness, users may specify the mode explicitly if they wish.
+The presence of the ``mode`` key also ensures that **freud** will not have to change its promises around mode deduction as additional query modes are added.
+
+
+Query Results
```

Comment:
This section repeats itself a fair amount. What are the reasons why a reader would want to know more about this? We need to be sure we're addressing those reasons and not just showing off how cool it is.